### PR TITLE
vimPlugins.neotest: add dependencies

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

[In the neotest README](https://github.com/nvim-neotest/neotest#installation), it is said that
- plenary.nvim is a "hard" dependency
- FixCursorHold.nvim is "recommended"

This patch adds both in the `neotest` derivation.
Should only plenary be actually included ?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 267847` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.neotest</li>
  </ul>
</details>

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -847,6 +847,10 @@ self: super: {
     dependencies = with self; [ plenary-nvim ];
   };
 
+  neotest = super.neorg.overrideAttrs {
+    dependencies = with self; [ FixCursorHold-nvim plenary-nvim ];
```
Comment (ambroisie) [inline pkgs/applications/editors/vim/plugins/overrides.nix:851]: Given that the README says it shouldn't be needed after https://github.com/neovim/neovim/pull/20198, I'd remove it.

**Review by ambroisie — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vim/plugins/overrides.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/overrides.nix]: Indeed !
Done.

**Review by ambroisie — APPROVED** [review]

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 2]
```diff
@@ -847,6 +847,10 @@ self: super: {
     dependencies = with self; [ plenary-nvim ];
   };
 
+  neotest = super.neorg.overrideAttrs {
```
Comment (ohlus) [inline pkgs/applications/editors/vim/plugins/overrides.nix:850]: Shouldn't it be `super.neotest.overrideAttrs`?
It looks like a copy paste error from the previous neorg attribute.

I think this is the reason why my nixvim config stopped working.

Comment (ambroisie) [reply pkgs/applications/editors/vim/plugins/overrides.nix:850]: Oh totally right, sorry I missed it in review 🙃.

Comment (ohlus) [reply pkgs/applications/editors/vim/plugins/overrides.nix:850]: I have just created #269063, if you could look at it, thanks :+1:

Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/overrides.nix:850]: ooops ! Sorry for that :/
Thanks for the quick fix @ambroisie !

Comment (ambroisie) [reply pkgs/applications/editors/vim/plugins/overrides.nix:850]: @ohlus is the one to thank here, I'm just approving the MR (no commit bit unfortunately).

**Pull request merged**